### ISSUE TITLE
[clsim] Frame cache never flushed if no light sources encountered (Trac #2402)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2394]
I3CLSimClientModule never flushes its frame cache it encounters 1 or more Q frames, but none contain work.

The underlying condition is an error itself (why run photon propagation with no work?), but the case should still be caught so the frames aren't lost. 

Migrated from https://code.icecube.wisc.edu/ticket/2402
```json
{
    "status": "closed", 
    "changetime": "2020-06-24T12:31:42", 
    "description": "I3CLSimClientModule never flushes its frame cache it encounters 1 or more Q frames, but none contain work.\n\nThe underlying condition is an error itself (why run photon propagation with no work?), but the case should still be caught so the frames aren't lost. ", 
    "reporter": "jvansanten", 
    "cc": "wtian, dxu", 
    "resolution": "fixed", 
    "_ts": "1593001902142004", 
    "component": "combo simulation", 
    "summary": "[clsim] Frame cache never flushed if no light sources encountered", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2020-01-16T18:47:30", 
    "milestone": "Autumnal Equinox 2020", 
    "owner": "jvansanten", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20200116T19:40:00: 
* jvansanten commented:

> In changeset 178497:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="178497"
> Flush frame cache when there was no work
> 
> Fixes #2394.
> ```


* jvansanten changed status from "new" to "closed"
* jvansanten changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20200624T12:31:42: 
* olivas changed milestone from "Vernal Equinox 2020" to "Autumnal Equinox 2020"
* olivas commented:

> Milestone Vernal Equinox 2020 deleted
